account-kit/core/src/actions/getAccount.ts

```diff
@@ -28,17 +31,33 @@ export type GetAccountParams<TAccount extends SupportedAccountTypes> =
  * @returns {GetAccountResult<TAccount>} The result which includes the account if found and its status
  */
 export const getAccount = <TAccount extends SupportedAccountTypes>(
-  { type }: GetAccountParams<TAccount>,
+  { type, accountParams }: GetAccountParams<TAccount>,
   config: AlchemyAccountsConfig
 ): GetAccountResult<TAccount> => {
   const accounts = config.store.getState().accounts;
   const chain = getChain(config);
   const account = accounts?.[chain.id]?.[type];
+  const accountConfig =
+    config.store.getState().accountConfigs[chain.id]?.[type];
+
+  if (type === "ModularAccountV2" && account?.status === "READY") {
```

Comment:
Why is the guard before the other guard, wouldn't running the guard on line 60 with just checking the existence of accountConfig be simpler?
Looking for the reasoning.

Reply:
great call-out! pushing a change now.